InstanceManager.js

```diff
@@ -557,70 +651,59 @@ class InstanceManager {
 
   /// use Current/ServersRegistered to find the corresponding
   /// instance metadata
-  resolveUUID(uuid) {
+  async resolveUUID(uuid) {
     const baseUrl = endpointToUrl(this.getAgencyEndpoint());
-    return this.rpAgency({
+    const [info] = await this.rpAgency({
       method: 'POST',
       uri: baseUrl + '/_api/agency/read',
       json: true,
       body: [['/arango/Current/ServersRegistered']]
-    }).then(([info]) => {
-      const servers = info.arango.Current.ServersRegistered;
-      let url = servers[uuid].endpoint;
-      return this.instances.filter(inst => inst.endpoint == url).shift();
     });
+    const servers = info.arango.Current.ServersRegistered;
+    let url = servers[uuid].endpoint;
+    return this.instances.filter(inst => inst.endpoint == url).shift();
   }
 
-  assignNewEndpoint(instance) {
+  async assignNewEndpoint(instance) {
     let index = this.instances.indexOf(instance);
     if (index === -1) {
       throw new Error("Couldn't find instance " + instance.name);
     }
 
-    var createNewEndpoint = () => {
-      return this.runner.createEndpoint().then(endpoint => {
-        if (endpoint != instance.endpoint) {
-          return endpoint;
-        } else {
-          return createNewEndpoint();
-        }
-      });
-    };
+    let endpoint;
+    do {
+      endpoint = await this.runner.createEndpoint();
+    } while (endpoint == instance.endpoint);
 
-    return createNewEndpoint().then(endpoint => {
-      [
-        'server.endpoint',
-        'agency.my-address',
-        'cluster.my-address'
-      ].filter(arg => {
-        index = instance.args.indexOf('--' + arg + '=' + instance.endpoint);
-        if (index !== -1) {
-          instance.args[index] = '--' + arg + '=' + endpoint;
-        }
-      });
-      return this.runner.updateEndpoint(instance, endpoint);
+    [
+      'server.endpoint',
+      'agency.my-address',
+      'cluster.my-address'
+    ].filter(arg => {
+      index = instance.args.indexOf('--' + arg + '=' + instance.endpoint);
+      if (index !== -1) {
+        instance.args[index] = '--' + arg + '=' + endpoint;
+      }
     });
+
+    return this.runner.updateEndpoint(instance, endpoint);
   }
 
   // beware! signals are not supported on windows and it will simply do hard kills all the time
   // use shutdown to gracefully stop an instance!
-  kill(instance) {
+  async kill(instance) {
     if (!this.instances.includes(instance)) {
       throw new Error("Couldn't find instance " + instance.name);
     }
 
     instance.process.kill('SIGKILL');
     instance.status = 'KILLED';
-    return new Promise((resolve, reject) => {
-      const check = () => {
-        if (instance.status !== 'EXITED') {
-          setTimeout(check, 50);
-        } else {
-          resolve();
-        }
-      };
-      check();
-    });
+    while (true) {
+      if (instance.status === 'EXITED') {
+        return;
+      }
+      await sleep(50);
+    }
```

Comment:
Maybe simply
```js
while (instance.status !== 'EXITED') {
  await sleep(50);
}
```
?

This would make it more obvious when the loop will hang.